### ISSUE TITLE
h2_ssl_proxy_call_creds_test flaky

### ISSUE THREAD

**Issue by dgquintas** [grpc/grpc#3240]
Running h2_ssl_proxy_call_creds_test under asan for ~50 iterations surfaces different types of flakes (possibly related): 

https://gist.github.com/dgquintas/8a00ff51d0062fa2907b

https://gist.github.com/dgquintas/dd41cf72d298f9154052

This at HEAD on both master and release-0.11

**Comment by jboeuf** [comment]
I cannot repro this one on master. I ran:
```
05:39:03 {master} ~/Development/grpc$ ./tools/run_tests/run_tests.py -l c -c asan -r h2_ssl_proxy_call_creds_test -n 100 -a 25
SUCCESS: All tests passed
```

**Comment by dgquintas** [comment]
-c tsan

**Comment by jboeuf** [comment]
Same thing:
```
05:55:41 {master} ~/Development/grpc$ ./tools/run_tests/run_tests.py -l c -c tsan -r h2_ssl_proxy_call_creds_test -n 100 -a 25
SUCCESS: All tests passed
```

**Comment by dgquintas** [comment]
hmm maybe leave it running with -S -n inf -a 100 ?

That will: stop upon failure (-S), run continuously (-n inf) and make your machine feel the pain the jenkins' workers feel (-a 100).

**Comment by jboeuf** [comment]
Actually the trace linked on this one looks very similar to #3286 for which
I just issued a fix (#3291). Let me know if you agree and we can probably
close it.

**Comment by dgquintas** [comment]
Ah, very likely. I think it's fine to close.
